taichi/llvm/llvm_context.cpp

```diff
@@ -107,6 +107,8 @@ llvm::Type *TaichiLLVMContext::get_data_type(DataType dt) {
     return llvm::Type::getInt32Ty(*ctx);
   } else if (dt->is_primitive(PrimitiveTypeID::u64)) {
     return llvm::Type::getInt64Ty(*ctx);
+  } else if (dt->is<CustomIntType>()) {
+    return llvm::Type::getInt32Ty(*ctx);
```

Comment:
I don't think this branch belongs here. `CustomIntType` should not be exposed to LLVM. It makes more sense to do the branching on the caller side of this function.